tests/devapps/NetCoreTestApp/Program.cs

```diff
@@ -42,30 +43,56 @@ public class Program
         private readonly static string ClientIdForConfidentialApp = "<enter id>";
 
         private readonly static string Username = ""; // used for WIA and U/P, cannot be empty on .net core
-        private readonly static string Authority = "https://login.microsoftonline.com/organizations/v2.0"; // common will not work for WIA and U/P but it is a good test case
+        private readonly static string Authority = "https://login.microsoftonline.com/organizations"; // common will not work for WIA and U/P but it is a good test case
         private readonly static IEnumerable<string> Scopes = new[] { "user.read" }; // used for WIA and U/P, can be empty
 
         private const string GraphAPIEndpoint = "https://graph.microsoft.com/v1.0/me";
 
         public static void Main(string[] args)
         {
+            // with discrete logging calls (current API)
+            IPublicClientApplication pca = PublicClientApplicationBuilder
+                .Create(ClientIdForPublicApp)
+                .WithAadAuthority(AadAuthorityAudience.AzureAdMultipleOrgs)
+                .WithLoggingCallback(Log)
+                .WithLoggingLevel(LogLevel.Verbose)
+                .WithEnablePiiLogging(true)                
+                .Build();
+
+            // with a logging config
+            IPublicClientApplication pca2 = PublicClientApplicationBuilder
+                .Create(ClientIdForPublicApp)
+                .WithAadAuthority(AadAuthorityAudience.AzureAdMultipleOrgs)
+                .WithLoggingConfig(
+                    LoggingConfig.Create(Log)
+                    .WithLogLevel(LogLevel.Verbose)
+                    .EnablePii(true)
+                    .Build())
+                .Build();
+
+            // or with logging config separated out
+            var loggingConfig = LoggingConfig.Create(Log)
```

Comment:
This misses the case where we want things like LogLevel, Pii Enable, etc to come from the config file.  We already have a PublicClientApplicationBuilder.CreateWithOptions() to support just that so the developer can have one json blob that loads up their entire app config (minus callbacks).  

Also, are you suggesting we do this for telemetry as well?  because telemetry also has a callback but it doesn't have a telemetry builder and there are no other options (for now) on telemetry.  so that makes things a bit consistent.

I like what you're doing here but I personally think this overcomplicates the API and some of our scenarios without adding enough value (e.g. developer needs to now know about the LoggingConfig class in that special case)

Reply:
Thinking more about this, I like the idea of having the one method for WithLogging(callback, enablePii, LogLevel).  Have the callback required and then have the other values there as optional parameters to configure it.  The other configurations don't make sense if you dont' specify a log callback.